packages/pigeon/lib/cpp_generator.dart

```diff
@@ -669,6 +669,8 @@ class CppSourceGenerator extends StructuredGenerator<CppOptions> {
                 .map((Class x) => x.name)
                 .contains(field.type.baseName)) {
           return '${hostDatatype.datatype}::FromEncodableList(std::get<EncodableList>($encodable))';
+        } else if (field.type.baseName == 'Object') {
```

Comment:
Nit: you can pull this out to the outer if/else if/else cascade since it doesn't use `hostDatatype`.